feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,537 @@
+    a given threshold. The PSI of a feature is an indication of the shift in its
+    distribution; a feature with high PSI might therefore be seen as instable.
+
+    In fields like Credit Risk Modelling, the elimination of features with high PSI
+    is frequent and usually required by the Regulator.
+
+    The calculation of the PSI requires to compare two distributions.
+    In DropHighPSIFeatures the approach implemented is the following:
+
+    - The dataframe is split in two parts and the PSI calculations are performed by
+    comparing the distribution of the features in the two parts.
+
+    - The user defines the columns of the dataframe according to which the dataframe
+    will de split. The column can contain numbers, dates and strings.
+
+    - The user defines a split fraction. This is used to define the sizes of the two
+    dataframe. Is the split fraction is 50%, the two dataframe will have about the same
+    size. It is not guaranteed that the size will exactly match because the observations
+    are grouped by their label (i.e. their values in the split columns) so all
+    observations with the same label will automatically belong to the same dataframe.
+
+    - If the option split_distinct_value is activated, the number of distinct values
+    is used to make the split. For example a feature with values: [1, 2, 3, 4, 4, 4]
+    a 50% split without split_distinct_value will split into two parts [1, 2, 3] and
+    [4, 4, 4] while if the option is set on, the split will be [1, 2] and [3, 4, 4, 4].
+
+    The PSI calculations are not symmetric. The switch_basis argument allows to
+    switch the role of the two dataframes in the PSI calculations.
+
+    The comparison of the distribution is done through binning. Two strategies are
+    implemented: equal_frequency and equal_width. These labels refer to two
+    discretisation implementation from the present package.
+
+    References:
+    https://scholarworks.wmich.edu/cgi/viewcontent.cgi?article=4249&context=dissertations
+
+    DropHighPSIFeatures suited for numerical variables and for discrete variables with
+    high cardinality. For categorical variables differences in distribution can be
+    investigated using a Chi-square or Kolmogorov-Smirnov test.
+
+    Parameters
+    ----------
+
+    split_col: string.
+        Label of the column according to which the dataframe will be split.
+
+    split_frac: float.
+        Ratio of the observations (when split_distinct is not activated) that goes
+        into the sub-dataframe that is used to determine the reference for the feature
+        distributions. The second sub-dataframe will be used to compare its feature
+        distributions to the reference ones.
+
+    split_distinct_values: boolean.
+        If set on, the split fraction does not account for the number of observations
+        but only for the number of distinct values in split_col.
```

Comment:
If I understand this correctly, we are giving the user multiple ways in which they can split the data. 

My question is: would someone really split the data other than with a date? from what I understand from PSI, it is aimed at determining shifts in the distribution in time. So if we do not have a time measurement to split the data, it kind of defeats the point. Let me know if I am wrong.

My second thought on this front: simple is better, or less is more. Giving the users too many options makes the class harder to use. So I would keep it as simple as possible. The less parameters the better.

Reply:
PSI is used to compute population shift and a popular approach is to use time; for example remove features that are shifted comparing the pre-Covid to the Covid period. However it can also be used in other cases, like when combining two similar portfolios to make a common model. An example I know is the combination of two "similar" portfolios to build a common Credit Risk  model. The combination is needed because of the low number of observation in the positive class. PSI is then used to remove features than have different distribution between the two portfolios.

Reply:
I agree with the idea of less is more. I guess the first we need to agree on is if we want to have only a split based on time or if we want to allow a more general approach (and maybe touch a larger audience).
I can live with both approaches as I am personally using mostly  split according to time so for me the choice is yours.

Reply:
Yes, makes sense, I see the point. 

Thank you for taking the time to explain. 

Let's do as you suggest. 

We will need to add a lot of info in the docs, probably with these examples you mention here to help users understand how to use this class. The examples are nice and clear.